strings/check_anagram.nim

```diff
@@ -0,0 +1,179 @@
+## Check Anagram
+## =============
+## wiki: https://en.wikipedia.org/wiki/Anagram
+##
+## Two words are anagrams if:
+## - They are made up of the same letters.
+## - The letters are arranged differently.
+## - The case of the characters in a word is ignored.
+## - They are not the same word (a word is not an anagram of itself).
+##
+## A word is any string of characters `{'A'..'Z', 'a'..'z'}`. Other characters,
+## including whitespace, numbers and punctuation, are considered invalid and
+## raise a `ValueError` exception.
+##
+## Note: Generate full doc with `nim doc --docinternal check_anagram.nim`
+
+runnableExamples:
+  doAssert "coder".isAnagram("credo")
+
+  doAssert not "Nim".isAnagram("Nim")
+
+  doAssert not "parrot".isAnagram("rapport")
+
+  let raisedException = try:
```

Comment:
here you could use `doAssertRaises(ValueError)`, see https://nim-lang.org/docs/assertions.html#doAssertRaises.t%2Ctypedesc%2Cuntyped

Reply:
Thanks, missed it.